doc/source/loggers.rst

```diff
@@ -13,24 +13,106 @@ by caproto, below. Because recalling the name is not always convenient, certain
 objects in caproto's Python API expose a ``log`` attribute, such as ``pv.log``,
 ``pv.context.log`` and ``pv.circuit_manager.log``.
 
+You could find python logging flow here.
+https://docs.python.org/2/_images/logging_flow.png
+
+
 Useful snippets
 ===============
 
-To turn on maximal logging (usually way too much information!) use
+Make sure you understand how level control loggging in logger.
+and handler. Both has method setLevel(...) which allow you do logger.setLevel(...) or
+handler.setLevel(...). You may only have one logger in your package. But, mutiple
+handlers could be added to one logger. Logger's level influence all handler. So the
+effective level is the intersection of logger's level and handler's level.
+
+
+Here INFO will effect
 
 .. code-block:: python
 
     import logging
-    logging.getLogger('caproto').setLevel('DEBUG')
+    logger = logging.getLogger('caproto')
+    logger.setLevel('DEBUG')
+    handler = logging.StreamHandler()
+    handler.setLevel('INFO')
+    logger.addHandler(handler)
+    logger.debug('This is debug message')
+    logger.info('This is info message')
+    logger.warning('This is warn message')
+
+.. code-block:: python
+
+    This is info message
+    This is warn message
+
+Here WARNING will effect
+
+.. code-block:: python
+
+    import logging
+    logger = logging.getLogger('caproto')
+    logger.setLevel('WARNING')
+    handler = logging.StreamHandler()
+    handler.setLevel('INFO')
+    logger.addHandler(handler)
+    logger.debug('This is debug message')
+    logger.info('This is info message')
+    logger.warning('This is warn message')
+
+.. code-block:: python
+
+    This is warn message
+
+To avoid confused level setting, we recommend leave logger's level 'NOTSET' and use
+handler's level domain independently.
 
 To log (non-batch) read/write requests and read/write/event responses in the
 threading and pyepics-compat clients:
 
 .. code-block:: python
 
    import logging
-   logging.getLogger('caproto.ch').setLevel('DEBUG')
+   logger = logging.getLogger('caproto')
+
+To add a stdout handler with level, default level is WARNING
+
+.. code-block:: python
+
+    from caproto._log import set_handler
+    std_handler = set_handler(level='DEBUG')
+    logger.addHandler(std_handler)
```

Comment:
This line isn't necessary. The `set_handler` function automatically adds the new handler to the `'caproto'` logger.

Reply:
I see. But my point is make set_handler play the identical role as logging.Streamhandler() definition which need explicit add later. This also make all handler adding follow same pattern. Is it gonna hurt current running code? Also keep it won't have duplicate handler in `logger.handlers` . Anyway, I'm open to delete it.